### PR TITLE
[Android] Enhance the scan of the games process

### PR DESCRIPTION
**Pull request by mahdihijazi**
- Move the operation into the background thread
- Show a dialog to indicate that the process is in progress

if you have a small set of games, there is a possibility that the dialog will not be noticeable at all.

Below video show the change:

[video](http://www.youtube.com/watch?v=VU57PU23CLY)

### CODE REVIEW

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/main/MainPresenter.java [hunk 1]
```diff
@@ -56,9 +57,35 @@ public boolean handleOptionSelection(int itemId)
 				return true;
 
 			case R.id.menu_refresh:
+				mView.showScanDialog();
 				GameDatabase databaseHelper = DolphinApplication.databaseHelper;
-				databaseHelper.scanLibrary(databaseHelper.getWritableDatabase());
```
Comment (degasus) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/main/MainPresenter.java]: I wonder will the old helper still be called somewhere at all?

**Comment by shuffle2** [comment]
Similar to the non Android builds, it is better ux to have a non intrusive scanning progress indicator...

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/main/MainPresenter.java (hunk 1 above)
Comment (mahdihijazi) [reply Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/main/MainPresenter.java]: yes, it is called inside the GameProvider content provider, which is by nature should be a blocking call, I can reuse the other one with Observable but it looks an overkill for other cases.

**Comment by mahdihijazi** [comment]
@shuffle2 @degasus is this option less intrusive?

[video](http://www.youtube.com/watch?v= veyHwOr2YI4)

**Comment by mahdihijazi** [comment]
- Use one scanLibrary helper method everywhere
- Use text view as status bar for tv and Snackbar for phones

**Comment by mahdihijazi** [comment]
@hackbar can you check this one

**Comment by hackbar** [comment]
Having a status bar at the bottom isn't good mobile UX, and it looks very strange. Given the choice, I'd prefer the 'loading' dialog.

But I think the right way to do this would be to hook to the Adapter, so as games are loaded new boxes appear. That might require a lot of database refactoring (which would be beneficial regardless).

**Comment by mahdihijazi** [comment]
@hackbar I agree that loading the games while scanning will be the best UX but as you said that probably will require more work and we can do that in the next step.

@MayImilae @degasus Regarding the UX part who can give us a final decision here?

**Comment by MayImilae** [comment]
Text just on the bottom like that is super ugly, and very un-android like! Is there an android accepted way of showing information like that?

Also, I'm fine with an intrusive pop up refreshing behavior if it only happens when a user presses the button. As long as it isn't automatic, I'm good!

**Comment by mahdihijazi** [comment]
@MayImilae I will restore the original dialog then again, and annoy people to merge it after that :)

**Comment by mahdihijazi** [comment]
@MayImilae I restored the dialog for the Android TV again

**Comment by mahdihijazi** [comment]
@degasus can you please take a look at the update and merge if everything is fine.

**Comment by hackbar** [comment]
This is all pretty complicated and kinda unweildy for not much benefit. I still think it would be better to fix this the right way, in the underlying adapter, rather than trying to glue more on top.

**Pull request closed without merging**